### PR TITLE
[Feature] Update E4 teams

### PR DESCRIPTION
**Pull request by Tempo-anon**
## What are the changes?
Updated elite four teams. Let me know if you have any comments, suggestions, or feedback

## Why am I doing these changes?
* Make them all fit the standard template of 4 signature Pokemon
* Try to allow more variation between Pokemon that fulfill similar roles
* Try to highlight Pokemon from their generation
* Try to make more unique teams

## What did change?
Updated teams for:
* Lorelei
* Bruno
* Agatha
* Will
* Koga
* Sidney
* Phoebe
* Aaron
* Flint
* Lucian
* Shauntal
* Hala
* Olivia
* Acerola
* Kahili
* Marnie
* Nessa
* Bea
* Allister
* Raihan
* Rika
* Hassel
* Lacey
* Drayton

## How to test the changes?
Override to floor 182

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [ ] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/data/trainer-config.ts [hunk 1]
```diff
@@ -876,58 +876,58 @@ export const signatureSpecies: SignatureSpecies = {
...
+  SHAUNTAL: [Species.COFAGRIGUS, Species.CHANDELURE, Species.GOLURK, Species.JELLICENT],
   MARSHAL: [Species.CONKELDURR, Species.MIENSHAO, Species.THROH, Species.SAWK],
   GRIMSLEY: [Species.LIEPARD, Species.KINGAMBIT, Species.SCRAFTY, Species.KROOKODILE],
   CAITLIN: [Species.MUSHARNA, Species.GOTHITELLE, Species.SIGILYPH, Species.REUNICLUS],
   MALVA: [Species.PYROAR, Species.TORKOAL, Species.CHANDELURE, Species.TALONFLAME],
   SIEBOLD: [Species.CLAWITZER, Species.GYARADOS, Species.BARBARACLE, Species.STARMIE],
   WIKSTROM: [Species.KLEFKI, Species.PROBOPASS, Species.SCIZOR, Species.AEGISLASH],
   DRASNA: [Species.DRAGALGE, Species.DRUDDIGON, Species.ALTARIA, Species.NOIVERN],
-  HALA: [Species.HARIYAMA, Species.BEWEAR, Species.CRABOMINABLE, Species.POLIWRATH],
+  HALA: [Species.HARIYAMA, Species.BEWEAR, Species.CRABOMINABLE, [Species.POLIWRATH, Species.POLIWRATH]],
```
Comment (Fontbane) [inline src/data/trainer-config.ts:902]: Their roles are WAY too different to put into one slot

**Comment by flx-sta** [comment]
> Try to allow more variation between Pokemon that fulfill similar roles

I was expecting it to have a pool of Pokémon that are randomly selected from with like 4 fixed ones

**Comment by Fontbane** [comment]
On Indigo:
- Lapras is Lorelei's ace, along with Jynx, so it should have a permanent slot. Sandslash is kinda extra, can be replaced with a [Dewgong, Cloyster]
- Not sure on the Golem over Onix, Onix is pretty well known for him whereas Golem was just in one LGPE battle.
- Muk is kind of a mainstay for Koga, whereas he only uses Tentacruel once in an LGPE rematch. I'd keep Weezing and Muk as separate slots over Tentacruel.
On Hoenn:
- I'd give Sidney a [Sharpedo, Crawdaunt], since he uses it in Emerald, which gives him some nice flexibility the other Hoenn E4s can't really afford.
On Sinnoh:
- I like the flex slot using his off type mons. Maybe stick Drifblim in there too.
On Unova:
- Nothing wrong here just rip Bisharp, bro is forgotten
On Alola:
- I made a joke about it before but I'm assuming you meant for Hala's second Poliwrath to be Primeape?
- Acerola never uses a Mimikyu in battle. I'd put Froslass instead, since I'm guessing the random slots would have a good chance to pull Mimikyu anyway.
On Galar:
- Since Allister and Bea aren't G-Maxing, I'd get rid of Machamp and Gengar tbh. Lets them use more Galar mons. If not, I'd shift Allister's [] to [Chandelure, Dusknoir] so he always has Runerigus. You could also put in Polteageist, since his gym is kinda themed after it and he uses it in his E4 match.
- Probably not for this PR, but what do you think about giving the final E4 members a mega/G-max? Would help Raihan differentiate from Drayton (who could get Mega Sceptile)
On the BBLeague:
- I think Lacey should always have Excadrill, it's her ace and ties back to her dad. Could put Galar bro in with Alcremie and Granbull, since two other E4s already have a Slowbro.

**Review by damocleas — APPROVED** [review]

**Comment by Tempo-anon** [comment]
> On Indigo:
> 
>     * Lapras is Lorelei's ace, along with Jynx, so it should have a permanent slot. Sandslash is kinda extra, can be replaced with a [Dewgong, Cloyster]

You're right about Lapras, it is pretty iconic for her and better than Dewgong. I do want to give her a Cloyster but ASandslash is also very unique and a more recent LGPE addition.

>     * Not sure on the Golem over Onix, Onix is pretty well known for him whereas Golem was just in one LGPE battle.
I believe the Onix would automatically evolve into Steelix. I actually like Golem/AGolem as an addition since those are Pokemon that nobody else really uses

>     * Muk is kind of a mainstay for Koga, whereas he only uses Tentacruel once in an LGPE rematch. I'd keep Weezing and Muk as separate slots over Tentacruel.
Koga's ace in GSC was Crobat so I definitely wanted to keep that. Ariados is an alternative to Venomoth and is also gen 2 representation. I like Tentacruel to give him some type variation since Weezing and Muk are both mono-poisons. Maybe something from his Forretress from GSC too?

> On Hoenn:
> 
>     * I'd give Sidney a [Sharpedo, Crawdaunt], since he uses it in Emerald, which gives him some nice flexibility the other Hoenn E4s can't really afford.
Great suggestion! I'll definitely do that. Sidney has a bunch of team flexibility compared to the other Hoenn E4 that all have repeat mons

> On Sinnoh:
> 
>     * I like the flex slot on Flint using his off type mons. Maybe stick Drifblim in there too.
Lopunny and Steelix are infamous on Flint. I did forget about Drifblim though, but Drifblim is a very common Pokemon in named trainer teams
> On Unova:
> 
>     * Nothing wrong here just rip Bisharp, bro is forgotten
Bisharp/Kingambit is on sooooo many named trainer teams
> 
> On Alola:
> 
>     * Acerola never uses a Mimikyu in battle. I'd put Froslass instead, since I'm guessing the random slots would have a good chance to pull Mimikyu anyway.
I went with Mimikyu since that's one of her Pokemon in Pokemon Masters Ex and she has one in the anime. 
> 
> On Galar:
> 
>     * Since Allister and Bea aren't G-Maxing, I'd get rid of Machamp and Gengar tbh. Lets them use more Galar mons. If not, I'd shift Allister's [] to [Chandelure, Dusknoir] so he always has Runerigus. You could also put in Polteageist, since his gym is kinda themed after it and he uses it in his E4 match.
I admit I did not play sw/sh 🤷 
>     * Probably not for this PR, but what do you think about giving the final E4 members a mega/G-max? Would help Raihan differentiate from Drayton (who could get Mega Sceptile)
Hmmm, could be!
> On the BBLeague:
> 
>     * I think Lacey should always have Excadrill, it's her ace and ties back to her dad. Could put Galar bro in with Alcremie and Granbull, since two other E4s already have a Slowbro.
I also did not play the SV DLC so I don't know these characters and why I'm asking for feedback. Lacey will have her Excadrill! Let me know if there's anything else I should change

**Review by brain-frog — APPROVED** [review]

**Pull request merged**